htdocs/config.inc

```diff
@@ -13,9 +13,11 @@ function logException(Exception $e) {
 			'E-Mail: '.$account->getEmail().EOL.EOL.'-----------'.EOL.EOL;
 	}
 	$message .= 'Error Message: '.$e->getMessage().EOL.EOL.'-----------'.EOL.EOL;
-	if($currMySQLError = mysql_error()) {
+	require_once(LIB . '/Default/SmrMySqlDatabase.class.inc');
+	$db = new SmrMySqlDatabase();
```

Comment:
Right now this potentially creates a db instance/connection when we're in an error case, which I'd rather avoid, I think this can probably be done via `global $db` and checking if there's already one available, if not's not globally available usually then using `class_exists` should do the trick.  I doubt it's the end of the world if we do, but it'd be preferable not to - if we do have to then we should at least try/catch it so we can still send the original message, like with `release_lock()` below

Reply:
Based on http://php.net/manual/en/mysqli.construct.php, it looks like the `mysqli` constructor is exception-free. Since an object is returned even if `mysqli(...)` fails, then `self::$dbConn (a class static) will be set and future calls to the `MySqlDatabase` constructor will succeed (because it will see that `self::$dbConn` exists and therefore skip the failing `mysqli` call. This allows us to get the error message as we expect.

On the other hand, if we were to use `global $db`, then we will actually miss the error message because the first (failing) call to the `MySqlDatabase` constructor never completes, and so `$db` won't exist in `logException`.

I believe the code is correct as-is, unless there is some case that you think I've forgotten.